### PR TITLE
Create Spurious symptoms as objects

### PR DESCRIPTION
**Pull request by RobertManningSmith**
This is part 1 of 2 of the new and tested spurious symptoms/ covid testing PR.

Note that the open pull requests 26 and 27 have been included here, hence the massive number of commits included in this.

This pull request introduces the spurious symptom framework as an object which is created for each person at a given rate. Once created this object will show an "exposed" behaviour node for a week and then revert to a "susceptible" behaviour node. From this point, existing nodes can remain dormant (susceptible) or flare up again.

There is now also a spurious symptom testing suit which is now working.

I should have stopped at just creating these spurious symptoms but i have gone on a bit to include covid testing. The covid testing files are probably subject to change and not worth spending to much time on for now.

### CODE REVIEW

**Comment by RobertManningSmith** [comment]
To help with review:

Files that have had a small change:

Pom xml - just changed a build requirement that caused the maven build to fail for java 8

Demography - created and used a ticks per year parameter
Movement behaviour - changed an import
Location - changed an import
CoronavirusBehaviourFramework - changed a name of a variable
Infection - changed an import
Bulkrun - changed the simulation object to have an extra boolean
ImportExport - changed an import
Params: created new parameters (not imported)
Runmosttravelled: changed the simulation object to have an extra boolean
Singlerun: changed the simulation object to have an extra boolean

All testing files except Covid spurious symptom/testing tests: changed simulation object to handle an extra boolean



Files that need a review:

Person: created properties that determine who has symptoms and who should be tested. Also created corresponding getters and setters

CoronavirusSpuriousSymptom: creates the spurious symptom object class

InfectiousBehaviourFramework: attempted to create a way to generalise disease behaviour frameworks. I didn't manage to do this actually so probably best to discuss

SpuriousSymptomBehaviourFramework: creates the behaviours exhibited by the spurious symptom object

CovidSpuriousSymptoms: handles the steppable functions which control the creation and removal of symptoms

CovidTesting: handles the covid testing administration

DiseaseTesting: an interface for general disease testing, implementation didn't seem to be the best, would be good to discuss

Logging: created an output file for covid testing

SpuriousSymptoms: generalised interface for symptoms

WorldBankCovid19Sim: just changes things to handle spurious symptoms and testing

CovidSpuriousSymptomTesting: test suite for covid symptom objects

CovidTestingTesting: test suite for covid testing

File: Protecs/src/main/java/uk/ac/ucl/protecs/objects/Person.java [hunk 1]
```diff
@@ -140,8 +140,18 @@ public static OCCUPATION getValue(String x) {
 	boolean recovered = false;
 	boolean hasCovid = false;
 	boolean hadCovid = false;
+	
+	// Covid testing properties
```
Comment (swise5) [inline Protecs/src/main/java/uk/ac/ucl/protecs/objects/Person.java:144]: as a placeholder; we discussed modularising and abstracting diseases here, especially in light of beginning to expand the set of diseases we're considering.

File: Protecs/src/main/java/uk/ac/ucl/protecs/objects/diseases/SpuriousSymptomBehaviourFramework.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+				// Use switch statement to clearly create conditional actions based on the current state of this person's symptoms 
+				String action = "";
+				// if this is there first time then they will have a time of creation and no recovery time set
+				if (symptom.timeRecovered == Double.MAX_VALUE) {
+					action = "initialSetUp";
+				}
+				if (time >= symptom.timeRecovered) {
+					action = "recover";
+				}
+				switch (action) {
```
[the remark this answers is not in the document]
Comment (swise5) [reply Protecs/src/main/java/uk/ac/ucl/protecs/objects/diseases/SpuriousSymptomBehaviourFramework.java:151]: changing object name from Title to Type after conversation

File: Protecs/src/main/java/uk/ac/ucl/protecs/sim/Params.java [hunk 3]
```diff
@@ -399,6 +397,10 @@ public void load_testing_locations(String testLocationsFilename) {
 				admin_zones_to_test_in.add(zone_to_test_in);
 			}
 			assert (admin_zones_to_test_in.size() > 0): "Number of admin zone to test in not loaded";
+			for (String location: admin_zones_to_test_in) {
+				assert (adminZoneNames.contains(location)): "Location to test in not found in model admin zones";
```
Comment (swise5) [inline Protecs/src/main/java/uk/ac/ucl/protecs/sim/Params.java]: let's add the name of the offending location for error checking!

File: Protecs/src/main/resources/dummy_n_tests_faulty.txt
Comment (swise5) [inline Protecs/src/main/resources/dummy_n_tests_faulty.txt:1]: @RobertManningSmith has opened an issue to create a folder to hold tests in resources directory (pending @swise5 figuring out what is happening with test/resources folder)

**Review by swise5 — COMMENTED** [review]
Great! A few minor tweaks as discussed.

**Review by swise5 — APPROVED** [review]
Looks great!

**Pull request merged**